log4j-api/src/main/java/org/apache/logging/log4j/spi/DefaultThreadContextMap.java

```diff
@@ -231,4 +231,21 @@ public boolean equals(final Object obj) {
         final Map<String, String> otherMap = other.getImmutableMapOrNull();
         return Objects.equals(map, otherMap);
     }
+
+    @Override
+    public Object save() {
+        return localMap.get();
```

Comment:
The `DefaultThreadContextMap` and `CopyOnWriteSortedArrayThreadContextMap` just return the underlying object, because they use immutable structures (an immutable map for the former and a frozen `StringMap` for the latter.

If you look at the implementation of `GarbageFreeSortedArrayThreadContextMap` it makes a copy of the map:

https://github.com/apache/logging-log4j2/blob/bf9c703cb7cfbce0b2b64bd6c64c4e03c43da1e0/log4j-api/src/main/java/org/apache/logging/log4j/spi/GarbageFreeSortedArrayThreadContextMap.java#L213-L217

It would still be possible to make this garbage-free if we use a recycler for the copied map.